src/types.rs

```diff
@@ -2038,17 +2040,56 @@ impl Rule {
     }
 }
 
-/// A [`Rule`] with holes; a sort of [`Rule`] template.
+/// A [`Rule`] with [`Hole`]s; a sort of [`Rule`] template.
+///
+/// See [`Context`] for more information.
 ///
 /// [`Context`]: enum.Context.html
 /// [`Rule`]: struct.Rule.html
+/// [`Hole`]: enum.Context.html#variant.Hole
+///
+/// # Examples
+///
+/// ```
+/// # use term_rewriting::{Signature, parse_rulecontext, RuleContext, Context, parse_context};
+/// let mut sig = Signature::default();
+///
+/// // Constructing a RuleContext manually.
+/// let left = parse_context(&mut sig, "A(B C [!])").expect("parse of A(B C [!])");
+/// let b = parse_context(&mut sig, "B [!]").expect("parse of B [!]");
+/// let c = parse_context(&mut sig, "C").expect("parse of C");
+///
+/// let r = RuleContext::new(left, vec![b, c]).unwrap();
+///
+/// // Constructing a RuleContext using the parser.
+/// let r2 = parse_rulecontext(&mut sig, "A(B C [!]) = B [!] | C").expect("parse of A(B C [!]) = B [!] | C");
+///
+/// assert_eq!(r, r2);
+/// ```
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub struct RuleContext {
+    /// The left hand side (lhs) of a RuleContext.
     pub lhs: Context,
+    /// The right hand sides (rhs) of a RuleContext.
     pub rhs: Vec<Context>,
 }
 impl RuleContext {
     /// Create a new `RuleContext` if possible.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, RuleContext, parse_context};
+    /// let mut sig = Signature::default();
+    ///
+    /// let left = parse_context(&mut sig, "A(B C [!])").expect("parse of A(B C [!])");
+    /// let b = parse_context(&mut sig, "B [!]").expect("parse of B [!]");
+    /// let c = parse_context(&mut sig, "C").expect("parse of C");
+    ///
+    /// let r = RuleContext::new(left, vec![b, c]).unwrap();
+    ///
+    /// assert_eq!(r.pretty(&sig), "A(B, C, [!]) = B [!] | C");
+    /// ```
```

Comment:
Can you please add examples showing conditions where `new` returns `None`?

Reply:
Can you also test the condition where the LHS is a lone variable (e.g. `x_ = B [!] x_`)?

These are the same two conditions that determine whether a `Rule` is valid, so it might also make sense to add a reference to `Rule::new` to the documentation here.